src/core/Akka.Tests.Shared.Internals/AkkaSpec.cs

```diff
@@ -136,30 +136,88 @@ protected T ExpectMsgPf<T>(string hint, Func<object, T> pf)
+                    ? aggregateException.Flatten().InnerExceptions[0]
+                    : ex;
+
+                var exceptionType = typeof(T);
+                return exceptionType == exception.GetType()
+                    ? (T)exception
+                    : throw new ThrowsException(exceptionType, exception);
+            }
 
-        protected void Intercept(Action actionThatThrows)
+            throw new ThrowsException(typeof(T));
+        }
+
+        /// <summary>
+        /// Ensure that an expected exception is thrown by the passed function value. The thrown exception must be an
+        /// instance of the type specified by the type parameter of this method. This method invokes the passed
+        /// function. If the function throws an exception that's an instance of the specified type, this method returns
+        /// void. Else, whether the passed function returns normally or completes abruptly with a different
+        /// exception, this method throws <see cref="ThrowsException"/>.
+        /// <para>
+        /// Also note that the difference between this method and <seealso cref="Intercept{T}"/> is that this method
+        /// does not return the expected exception, so it does not let you perform further assertions on that exception.
+        /// It also indicates to the reader of the code that nothing further is expected about the thrown exception
+        /// other than its type. The recommended usage is to use <see cref="AssertThrows{T}"/> by default,
+        /// <seealso cref="Intercept{T}"/> only when you need to inspect the caught exception further.
+        /// </para>
+        /// </summary>
+        /// <param name="actionThatThrows">The action that should throw the expected exception</param>
+        /// <exception cref="ThrowsException">If the passed action does not complete abruptly with an exception that's an instance of the specified type.</exception>
+        protected void AssertThrows<T>(Action actionThatThrows) where T : Exception
         {
             try
             {
                 actionThatThrows();
             }
-            catch(Exception)
+            catch (Exception ex)
             {
-                return;
+                var exception = ex is AggregateException aggregateException
+                    ? aggregateException.Flatten().InnerExceptions[0]
+                    : ex;
+
+                var exceptionType = typeof(T);
+                if (exceptionType == exception.GetType())
+                    return;
+
+                throw new ThrowsException(exceptionType, exception);
             }
-            throw new ThrowsException(typeof(Exception));
+
+            throw new ThrowsException(typeof(T));
         }
-        
-        protected async Task InterceptAsync(Func<Task> asyncActionThatThrows)
+
+        [Obsolete("Use AssertThrows instead.")]
+        protected void Intercept(Action actionThatThrows)
```

Comment:
Keeping this one since there are a bunch of tests using it, as a catch-all-exceptions, which is dubious at best.